sdks/python/apache_beam/runners/worker/bundle_processor.py

```diff
@@ -1243,17 +1243,28 @@ def create(factory, transform_id, transform_proto, serialized_fn, consumers):
 def create(*args):
 
   class PairWithRestriction(beam.DoFn):
-    def __init__(self, fn, restriction_provider):
+    def __init__(self,
+                 fn,
+                 restriction_provider,
+                 watermark_estimator_provider=None):
       self.restriction_provider = restriction_provider
-
+      self.watermark_estimator_provider = watermark_estimator_provider
     # An unused window is requested to force explosion of multi-window
     # WindowedValues.
     def process(
         self, element, _unused_window=beam.DoFn.WindowParam, *args, **kwargs):
       # TODO(SDF): Do we want to allow mutation of the element?
       # (E.g. it could be nice to shift bulky description to the portion
       # that can be distributed.)
-      yield element, self.restriction_provider.initial_restriction(element)
+      initial_restriction = self.restriction_provider.initial_restriction(
+          element)
+      if self.watermark_estimator_provider:
+        yield (element,
```

Comment:
We also need to emit (initial_restriction, initial_estimator_state) here to make sure `get_restriction_coder` works,